recipes/agent.rb

```diff
@@ -1,5 +1,15 @@
 apt_update
 package 'docker.io'
+service 'docker' do
```

Comment:
can the service be added after the template to save the `notifies` or is there any benefit in starting it first and restart just after deploying the configuration?

Reply:
In general yes, I try to order recipes with package, config files, service.

Due to the development order I didn't do that here and should. But we can't remove the notification because we want to make sure that a change to the config file triggers a restart if the docker service is already running.

Reply:
Services have been moved below configuration in the latest patch.